File: Client/src/app/brackets/bracket.ts
import { FighterDto } from '../fighters/fighter';
import { Page } from '../shared/models/page';

export interface CreateBracketDto {
  competitionId: string;
  name: string;
}

export interface Position {
  key: number;
  value: string;
}

export interface BracketDto {
  id: string;
  competitionId: string;
  name: string;
  fighters: FighterDto[];
  positions: Position[];
}

export class Round {
  constructor(public fighters: (FighterDto | null)[]) {}
}

export interface ShiftPage extends Page<BracketDto> {}

File: API.Data/Entities/Bracket.cs
namespace Data.Entities;

public class Bracket
{
    public string? Id { get; set; }
    public Competition? Competition { get; set; }
    public string? CompetitionId { get; set; }
    public string? Name { get; set; }

    public ICollection<Fighter> Fighters { get; set; } = [];
    public ICollection<Position> Positions { get; set; } = [];
}